website/docs/r/workloads_sap_three_tier_virtual_instance.html.markdown

```diff
@@ -0,0 +1,546 @@
+
+---
+
+A `virtual_machine_configuration` block supports the following:
+
+* `image` - (Required) An `image` block as defined below. Changing this forces a new resource to be created.
+
+* `os_profile` - (Required) An `os_profile` block as defined below. Changing this forces a new resource to be created.
+
+* `virtual_machine_size` - (Required) The size of the Virtual Machine. Changing this forces a new resource to be created.
+
+---
+
+An `image` block supports the following:
+
+* `offer` - (Required) Specifies the offer of the platform image or marketplace image used to create the virtual machine. Changing this forces a new resource to be created.
+
+* `publisher` - (Required) The publisher of the Image. Possible values are `RedHat` and `SUSE`. Changing this forces a new resource to be created.
+
+* `sku` - (Required) The SKU of the Image. Changing this forces a new resource to be created.
+
+* `version` - (Required) Specifies the version of the platform image or marketplace image used to create the virtual machine. Changing this forces a new resource to be created.
+
+---
+
+An `os_profile` block supports the following:
+
+* `admin_username` - (Required) The name of the administrator account. Changing this forces a new resource to be created.
+
+* `ssh_private_key` - (Required) The SSH public key that is used to authenticate with the Virtual Machine. Changing this forces a new resource to be created.
+
+* `ssh_public_key` - (Required) The SSH private key that is used to authenticate with the Virtual Machine. Changing this forces a new resource to be created.
+
+---
+
+A `three_tier_configuration` block supports the following:
+
+* `app_resource_group_name` - (Required) The name of the application Resource Group where SAP system resources will be deployed. Changing this forces a new resource to be created.
+
+~> **Note:** While creating an SAP Three Tier Virtual Instance, the service would provision the extra SAP system/component in the `app_resource_group_name` that isn't defined in the HCL Configurations. At this time, if the `app_resource_group_name` is different from the Resource Group where SAP Three Tier Virtual Instance exists, you can set `prevent_deletion_if_contains_resources` to `false` to delete all resources defined in the HCL Configurations and the resources created in the `app_resource_group_name` with `terraform destroy`. However, if the `app_resource_group_name` is the same with the Resource Group where SAP Three Tier Virtual Instance exists, some resources such as the subnet defined in the HCL Configurations cannot be deleted with `terraform destroy` since the resources defined in the HCL Configurations are being referenced by the SAP system/component. In this case, you have to manually delete the SAP system/component before deleting the resources in the HCL Configurations.
```

Comment:
```suggestion
~> **Note:** While creating an SAP Three Tier Virtual Instance, the service will provision the extra SAP systems/components in the `app_resource_group_name` that are not defined in the HCL Configuration. At this time, if the `app_resource_group_name` is different from the Resource Group where SAP Three Tier Virtual Instance exists, you can set `prevent_deletion_if_contains_resources` to `false` to delete all resources defined in the HCL Configurations and the resources created in the `app_resource_group_name` with `terraform destroy`. However, if the `app_resource_group_name` is the same with the Resource Group where SAP Three Tier Virtual Instance exists, some resources, such as the subnet defined in the HCL Configuration, cannot be deleted with `terraform destroy` since the resources defined in the HCL Configuration are being referenced by the SAP system/component. In this case, you have to manually delete the SAP system/component before deleting the resources in the HCL Configuration.
```

Reply:
Updated